orchagent/fdborch.cpp

```diff
@@ -174,41 +173,155 @@ bool FdbOrch::storeFdbEntryState(const FdbUpdate& update)
     }
 }
 
+
+/*
+Handles the SAI_FDB_EVENT_FLUSHED notification recieved from syncd
+clears stateDb and decrements corresponding internal fdb counters
+Ref: https://github.com/opencomputeproject/SAI/blob/master/inc/saifdb.h#L223
+*/
+void FdbOrch::handleSyncdFlushNotif(const sai_object_id_t& bv_id,
+                                    const sai_object_id_t& bridge_port_id,
+                                    const string& fdb_type,
+                                    const MacAddress& mac)
+{
+    MacAddress flush_mac(CONSOLIDATED_FLUSH_MAC);
+
+    /* Populate Queue */
+    vector<FdbUpdate> updates;
```

Comment:
Please avoid storing it in vector. The Port is a big Class and if there are 10K entries on a port for which flush it called this vector will spike the memory.
Instead call the storeFdbEntryState ,notify and ref count decrement while you iterate through the entries.